src/pages/InstructorStudentLanding.js

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import { useState, useEffect } from 'react';
+import CourseCard from '../components/CourseCard'
+import { Button, Card } from "react-bootstrap"
+import courseData from "./data/courseData.json"
+import apiUtil from '../utils/apiUtil'
+
+function InstructorStudentLanding(props) {
+
+    // Will implement ability to hide/show once we figure out best way to modify the
+    // data in the JSON with backend team
+    //const [showHidden, setShowHidden] = useState(false)
+
+    //user information
+    const userId = localStorage.getItem("id") // TODO: store in Redux
+
+    //course information
+    const [ studentCourses, setStudentCourses] = useState([])
+    const [ instructorCourses, setInstructorCourses] = useState([])
+
+
+    const [ editToggle, setEditToggle ] = useState(false)
+    const [ editCourse, setEditCourse ] = useState(false)
+
+    const controller = new AbortController()
+
+    useEffect( () => {
+        
+            async function populateCourses(){
+                
+                //get the courses for the user
+                const resp = await apiUtil('get', `/courses/`)
+                if (resp.status === 200) {
+                    setStudentCourses(resp.data.studentCourses)
+                    setInstructorCourses(resp.data.teacherCourses)
+                    
+                }
+                else {
+                    console.log(resp.data.error)
+                }
+                
+                //get the role from enrollment 
+
+            }
+            populateCourses()
+
+        }, [])
+
+        //student and teacher
+        if(studentCourses.length > 0 && !editToggle && instructorCourses.length > 0){
```

Comment:
I'm a little confused by the conditionals here if I'm being completely honest. What's edit toggle? And then also, do you think there is a cleaner way to minimize the code that's being repeated in each of the blocks following the conditional? Say, embed the conditions in the JSX? I trust it works, just a matter of making it easier to read and understand (and shorter 😄 )

Reply:
I honestly have no idea what edit toggle is, I was trying to preserve the original logic while also adding student vs teacher functionality to one page lol. I'll mess with the edit toggle to actually see what it does. I'll look into embedded conditions  as well and how they work, I originally had it in two separate if's, but I could only return one thing so it never got to the other one.